document-portal/document-portal.c

```diff
@@ -286,20 +286,22 @@ portal_delete (GDBusMethodInvocation *invocation,
 }
 
 static char *
-do_create_doc (struct stat *parent_st_buf, const char *path, gboolean reuse_existing, gboolean persistent)
+do_create_doc (struct stat *parent_st_buf, const char *path, gboolean reuse_existing, gboolean persistent, gboolean directory)
```

Comment:
Should we consider introducing flags here ? The booleans start to pile up

Reply:
Possibly. However we have several flags with these value, but most also have some other values. It might be pretty confusing which set of flags go where too.